types/src/transaction/authenticator.rs

```diff
@@ -272,7 +272,10 @@ impl FromStr for AuthenticationKey {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self> {
-        assert!(!s.is_empty());
```

Comment:
what.. who added this...

Reply:
can you add an testcase to make sure this don't appear again?

Reply:
it does not matter anymore after changed to use ensure, can't avoid people use assert with test, they will only add new use case.

I guess a lint rule will be better.

Reply:
anyway,  test added